### PR TITLE
apkg: init at 0.4.0

### PR DESCRIPTION
**Pull request by vcunat**
Add `apkg`: Upstream packaging automation tool
https://pkg.labs.nic.cz/pages/apkg

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] N/A Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] N/A Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 182362` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>apkg</li>
  </ul>
</details>

File: pkgs/tools/package-management/apkg/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    setuptools       # required by minver
+    toml             # config files
+  ];
+
+  makeWrapperArgs = [ # deps for `srcpkg` operation for other distros; could be optional
+    "--prefix" "PATH" ":" (lib.makeBinPath [ gitMinimal rpm dpkg fakeroot ])
+  ];
+
+  checkInputs = with python3Packages; [ pytest ];
+  checkPhase = "py.test"; # inspiration: .gitlab-ci.yml
```
Comment (smancill) [inline pkgs/tools/package-management/apkg/default.nix:37]: Missing the `runHook preCheck` and `runHook postCheck`.

Comment (vcunat) [reply pkgs/tools/package-management/apkg/default.nix]: OK, done.  It probably won't really matter here, but it's certainly cleaner to add those.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 182362` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>apkg</li>
  </ul>
</details>

**Review by smancill — APPROVED** [review]

**Pull request merged**